<launch>
	<!-- 是否开启rviz -->
	<arg name="rviz" default="true" />
	<!-- 设置输出结果保存路径 -->
	<arg name="time_file" default=""/>	
	<arg name="out_path" default=""/>	
	<arg name="out_origin_path" default=""/>
	<arg name="pose_log" default="false"/>
	<arg name="pose_log_file" default=""/>
	<arg name="cluster_out_file" default=""/>
	<arg name="time_breakdown_file" default=""/>

	<!-- 加载YAML文件(动态物体检测的参数) -->
	<rosparam command="load" file="$(find m_detector_noted)/config/kitti/kitti1.yaml" />
	<!-- 加载输出结果保存路径 -->
	<!-- 注意:如果YAML文件存在同名参数,launch文件参数优先级更高 -->
	<param name="dyn_obj/out_file" type="string" value="$(arg out_path)" />
	<param name="dyn_obj/out_file_origin" type="string" value="$(arg out_origin_path)" />
	<param name="dyn_obj/time_file" type="string" value="$(arg time_file)" />
	<param name="dyn_obj/pose_log" type="bool" value="$(arg pose_log)" />
	<param name="dyn_obj/pose_log_file" type="string" value="$(arg pose_log_file)" />
	<param name="dyn_obj/cluster_out_file" type="string" value="$(arg cluster_out_file)" />
	<param name="dyn_obj/time_breakdown_file" type="string" value="$(arg time_breakdown_file)" />

	<!-- 运行动态物体检测 -->
	<node pkg="m_detector_noted" type="dynfilter_noted" name="dynfilter_noted" output="screen" />
	<!-- 若开启rviz,则运行rviz节点 -->
	<group if="$(arg rviz)">
		<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find m_detector_noted)/rviz/demo.rviz" />
	</group>

  	  

</launch>
